app/contacts/contact_gist.py

```diff
@@ -51,7 +51,8 @@ async def handle_beacons(self, beacons):
         Handles various beacons types (beacon and results)
         """
         for beacon in beacons:
-            beacon['contact'] = self.name
+            if 'contact' not in beacon:
+                beacon['contact'] = self.name
```

Comment:
beacon['contact'] = beacon.get('contact', self.name) would be an alternate condensed way of writing this

Reply:
fixed!